packages/snjs/lib/protocol/collection/notes_display_criteria.ts

```diff
@@ -0,0 +1,208 @@
+import { SortDirection } from './item_collection';
+import { CollectionSort } from '@Lib/protocol/collection/item_collection';
+import { SNTag } from './../../models/app/tag';
+import { SNPredicate } from './../../models/core/predicate';
+import { ContentType } from './../../models/content_types';
+import { ItemCollection } from './item_collection';
+import { SNNote } from './../../models/app/note';
+import { SNSmartTag } from './../../models/app/smartTag';
+
+type SearchQuery = {
+  query: string
+  /** Whether to search the bodies of protected notes (as opposed to just title) */
+  protectedBodySearch: boolean
+}
+
+export class NotesDisplayCriteria {
+
+  public searchQuery?: SearchQuery
+  public tags?: SNTag[];
+  public includePinned = true;
+  public includeTrashed = false;
+  public includeArchived = false;
+  public sortProperty?: CollectionSort;
+  public sortDirection?: SortDirection;
+  /**
+   * Once all results are calculated, run them through this additional
+   * filter to determine whether they should be displayed.
+   */
+  public subFilter?: (element: any) => boolean;
+
+  static CreateCriteria(
+    initializer: (criteria: NotesDisplayCriteria) => void
```

Comment:
The only issue with that is that you can't have a one-liner, in terms of function call. You first have to instantiate another object, mutate it (which trains the caller that it's ok to mutate a NotesDisplayCriteria), then pass it into the function.

Reply:
With `Partial<NotesDisplayCriteria>` you can do this:

```ts
NotesDisplayCriteria.Create({
  includePinned: true
});
```

The object doesn't have to be an instance of `NotesDisplayCriteria`, it just has to respond identically to its property calls. See how only the last call in this example is invalid: https://www.typescriptlang.org/play?#code/MYGwhgzhAEAqAWBLAdgc2gbwFDWgBwFcAjERYaAMwHsroBeSsECAUwG4d9jTyiwAnetAAu-AuywBfLFgoFkwYYirIRYANYsICFKgAUwpGgBc0AAoClTADw60APgCUmTsBUQqIFgDoQVfYa63tRUjhzSWMIaWnb6GJJhkdHaRnGcIaai4lKJUZopunrYuHz8mWIsORx5MalFnKWmAEQA7ioA5MLQbgC2eIheTVVAA

In the end both approaches look kind of similar on the calling site and I don't hold a strong opinion either way, especially since this pattern also exists in other parts of the app. Just wanted to throw this at you in case you thought going the "pass an object" approach was impossible with TypeScript :)